### DISCUSSION TITLE
New probing features

### DISCUSSION THREAD

**Discussion by fra589** [fra589/cn5X#26] [Ideas]
Hi all,  
  
The work is in progress to add new probe function.

New user interfaces are almost finished:

![image](https://user-images.githubusercontent.com/19529154/107114553-f74e7880-6866-11eb-99da-f3c05647c7b7.png)
  
![image](https://user-images.githubusercontent.com/19529154/107114670-dc303880-6867-11eb-835c-7ad3af2d84ff.png)

  
And the Z probing function starts to work without too many errors :sweat_smile: 
  
At the same time, I add some additional menus and functions to manage the different systems of offsets and coordinates...

![image](https://user-images.githubusercontent.com/19529154/107114868-64fba400-6869-11eb-83b6-8a7d130a055f.png)

@++;  
Gauthier.

**Comment by HLex79** [comment]
Hi Gauthier,

wow very nice! I really need to try cn5x, it looks great!
I am currently rebuilding my CNC and adding an A axis, from grbl on uno to your version on mega!
I have to learn Freecad from scratch as F360 does not allow me to do multi axis, man is that complicated!

Do you have some good learning resources?

Thanks a lot for your magic work!

Best,

Lex.

**Reply by arlimbad222** [reply]
Hi @HLex79 

Just code for three axis and download software called gcode ripper it works fine with mega 5x just simple convert y moves to a .

**Reply by arlimbad222** [reply]
https://youtube.com/channel/UCOfz0ZG9viMBAumjTVag9Vw


This is my youtube channel link you can watch videos of my machine I use fusion 360 to create nc file for 3axis grbl and then go to gcode ripper and boom you are all set for rotary axis .

**Reply by HLex79** [reply]
Hi @arlimbad222,

I already read about Gcode ripper and I understand what it does/how it works. Nevertheless I intend to machine things like threads or keyways, so I will need the rotary axis code generated during tool path computation in the cad/cam phase.
Holy smokes, fusion 360 belongs to the past and now Freecad is completely new to me.... Lots of YouTube tutorials to come 
:-)
Thanks for your input!

Best,

Lex

**Reply by arlimbad222** [reply]
Hi @HLex79 

I can understand but I also find some free solutions to make a rotary tool path but I can't find anything & currently using ripper . if you find any open source solutions please give a links . 

Thankyou .

**Reply by fra589** [reply]
Hi @HLex79,  

> Do you have some good learning resources?  

I advise you to take a look on the Sliptonic Youtube channe:l https://www.youtube.com/c/sliptonic/videos
Sliptonic is one of the major developer of the Path module of FreeCAD.  
  
@++;  
Gauthier.

**Reply by fra589** [reply]
Hi @arlimbad222,  
  
> if you find any open source solutions please give a links
  
Here is the link: https://www.freecadweb.org/ 😃   

In fact, the Path module is in very active development a,d you need to use the latest dev. version to have the benefit of those new features like this: https://forum.freecadweb.org/viewtopic.php?f=15&t=36773  

The 4th axis is at his beginning cf. the Path module doc here: https://wiki.freecadweb.org/Fourth_Axis
I think that Sliptonic made a video for this, but I can't remember the link.

@++;  
Gauthier.

**Reply by HLex79** [reply]
Hi @fra589,

Gauthier, I sweep the net for informations. I am currently running the « normal » grbl on my mega and am upgrading to an a-axis. Mostly i use bCNC to run my codes, is this suitable to incorporate x y z and a commands or do you suggest something different?
I think i will need all my axis to stay powered in order to keep positional accuracy, also to be able to run normal xyz stuff.
I may be in the wrong issue topic, if so sorry.
This is kinda driving me nuts as there is no real source of complete informations.
Thanks for all hints!

Best,

Lex.

**Reply by fra589** [reply]
Hello,

I wrote cn5X++ because at the time when I started to create my CNC machine with more than 3 axes, there was no GCode sender capable of this.
Today, some senders have evolved and it seems that several of them know how to manage 4 axes and perhaps more. Personally, I don't know them since I only use cn5X++ for my milling machine.

Indeed, here is not the right place to ask this question :-) you will surely have better luck here: https://www.facebook.com/groups/238828216699975/

@ ++;
Gauthier.

**Reply by HLex79** [reply]
Hi,

You are absolutely right; i will restrict my questions to your awesome software that I will try in a not far away future 👍🏻 Thanks a lot for all of your efforts on all of this!
You know, when you are starting off, like a rookie like me, with no knowledge in coding, it is a huge unknown space difficult to understand.
Also; as i am really not willing to pay hundreds of dollars a year for CAD/CAM stuff, the learning curve is steep.
Figured out that i will need a lot of patience 😜
Thanks for everything!

Lex

**Comment by fra589** [comment]
Hi all,  
  
New version 0.8 pushed now:
- with full XYZ probing feature,
- new G92 dialog box functionality, 
- new dialogs for G28/G29 predefined positions...  

And many other details optimized or corrected.
  
@++;  
Gauthier.
